app/bundles/PageBundle/Entity/HitRepository.php

```diff
@@ -147,6 +148,11 @@ public function getHitCountForSource($source, $sourceId = null, $fromDate = null
      */
     public function getEmailClickthroughHitCount($emailIds, \DateTime $fromDate = null, $code = 200)
     {
+        $cacheStorageHelper = new CacheStorageHelper(CacheStorageHelper::ADAPTOR_FILESYSTEM);
+        $hitsCached         = $cacheStorageHelper->get('emails_list');
```

Comment:
I think you need to create unique names for every set of parameters, to store and read cache properly.

Reply:
emails_list is an unique name for cache, i verified it, and i didn't find nothing in ```app/bundles/EmailBundle/Controller/AjaxController.php``` and ```app/bundles/EmailBundle/Assets/js/email.js``` about caching.